src/test/java/dk/cngroup/trainings/spring/springassignment/controller/CareTakerControllerTest.java

```diff
@@ -0,0 +1,226 @@
+
+		mockMvc.perform(post("/careTakers")
+				.content(martinString)
+				.contentType(MediaType.APPLICATION_JSON))
+				.andExpect(status().isOk())
+				.andExpect(jsonPath("$.name", startsWith("Martin")));
+	}
+
+	@Test
+	public void testAddCareTakerWithEmptyName() throws Exception {
+		// Sad path
+		CareTaker martin = new CareTaker(1L, "                    ");
+		ObjectMapper objectMapper = new ObjectMapper();
+		String martinString = objectMapper.writeValueAsString(martin);
+
+		mockMvc.perform(post("/careTakers")
+				.content(martinString)
+				.contentType(MediaType.APPLICATION_JSON))
+				.andExpect(status().isBadRequest());
+	}
+
+	@Test
+	public void testGetCareTakerByExistingId() throws Exception {
+		// Happy path
+		CareTaker martin = new CareTaker(1L, "Martin Dobias");
+
+		Mockito.when(careTakerService.getCareTakerById(any(Long.class)))
+				.thenReturn(Optional.ofNullable(martin));
+
+		mockMvc.perform(get("/careTakers/10"))
+				.andExpect(status().isOk())
+				.andExpect(jsonPath("$.name", startsWith("Martin")));
+	}
+
+	@Test
+	public void testGetCareTakerByNonExistingId() throws Exception {
+		// Sad path
+		Mockito.when(careTakerService.getCareTakerById(any(Long.class)))
+				.thenReturn(Optional.empty());
+
+		mockMvc.perform(get("/careTakers/10"))
+				.andExpect(status().isNotFound());
+	}
+
+	@Test
+	public void testUpdateCareTakerByExistingIdAndValidCareTaker()
+			throws Exception {
+		// Happy path
+		CareTaker updatedJohn = new CareTaker(7L, "John Cina");
+
+		ObjectMapper objectMapper = new ObjectMapper();
+		String updatedJohnString = objectMapper.writeValueAsString(updatedJohn);
+
+		Mockito.when(careTakerService.getCareTakerById(any(Long.class)))
+				.thenReturn(Optional.ofNullable(careTakers.get(3)));
+		Mockito.when(careTakerService.addCareTaker(any(CareTaker.class)))
+				.thenReturn(Optional.ofNullable(updatedJohn));
+		Mockito.when(careTakerService.updateCareTakerById(any(Long.class), any(CareTaker.class)))
+				.thenReturn(Optional.ofNullable(updatedJohn));
+
+		mockMvc.perform(put("/careTakers/4")
+				.content(updatedJohnString)
+				.contentType(MediaType.APPLICATION_JSON))
+				.andExpect(status().isOk())
+				.andExpect(jsonPath("$.name", endsWith("Cina")));
+	}
+
+	@Test
+	public void testUpdateCareTakerByNonExistingIdAndValidCareTaker()
+			throws Exception {
+		// Sad path
+		CareTaker updatedJohn = new CareTaker(7L, "John Cina");
+
+		ObjectMapper objectMapper = new ObjectMapper();
+		String updatedJohnString = objectMapper.writeValueAsString(updatedJohn);
+
+		Mockito.when(careTakerService.getCareTakerById(any(Long.class)))
+				.thenReturn(Optional.empty());
+		Mockito.when(careTakerService.updateCareTakerById(any(Long.class),
+				any(CareTaker.class)))
+				.thenReturn(Optional.empty());
+
+		mockMvc.perform(put("/careTakers/9")
+				.content(updatedJohnString)
+				.contentType(MediaType.APPLICATION_JSON))
+				.andExpect(status().isNotFound());
+	}
+	/*
```

Comment:
this comment is about updating but is before deleting

Reply:
The position of the comment replace the position of the unneeded test for the same reason as in the answer for the previous question.